Readability.js

```diff
@@ -226,6 +234,30 @@ Readability.prototype = {
   },
 
   /**
+   * Removes the id="" and class="" attribute from every element in the given
+   * subtree, except those that match IDS_TO_PRESERVE and CLASSES_TO_PRESERVE.
+   *
+   * @param Element
+   * @return void
+   */
+  _cleanIDsAndClasses: function(node) {
+    if (node.nodeType != Node.ELEMENT_NODE) {
+      return;
+    }
+
+    if (node.id && this.IDS_TO_PRESERVE.indexOf(node.id) == -1) {
+      node.removeAttribute("id");
+    }
+    if (node.className && this.CLASSES_TO_PRESERVE.indexOf(node.className) == -1) {
+      node.removeAttribute("class");
+    }
+
+    for (node = node.firstChild; node; node = node.nextSibling) {
```

Comment:
Nit: it would be more efficient to use `firstElementChild` and `nextElementSibling` and then omit the `nodeType` check. :-)